src/components/DelegationNode/DelegationNode.tsx

```diff
@@ -283,16 +290,41 @@ class DelegationNode extends React.Component<Props, State> {
       }'`,
     })
 
-    Promise.chain(
-      hashes.map((hash: string, index: number) => () => {
+    const firstAttestation = await Attestation.query(hashes[0])
+
+    if (firstAttestation === null) {
+      throw SDKErrors.ERROR_NOT_FOUND('Attestation not on chain')
+    }
+
+    const steps = await DelegationsService.checkTraversalStepsToParent(
+      selectedIdentity.identity,
+      firstAttestation
+    )
+
+    await Promise.chain(
```

Comment:
I didn't know this existed! how does this compare to `Promise.all()` ?

Reply:
It has an option to continue through the promises even after a promise fails for whatever reason. 

If three promises are chained and the middle one throws an error it won't stop and throw and error straight away. It will continue to the next one and check if it is a success or not, then it will parse the successes and failures to a follow-up `then` to show errors or successful outcomes. 

During the function, it will complete the action if successful.


I hope I explained it well enough XD